x/yarpctest/core.go

```diff
@@ -0,0 +1,82 @@
+// Copyright (c) 2017 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package yarpctest
+
+import (
+	"fmt"
+
+	"github.com/stretchr/testify/assert"
+	"go.uber.org/multierr"
+	"go.uber.org/yarpc/x/yarpctest/api"
+)
+
+// Action is the interface for applying actions (Requests) in tests.
+type Action api.Action
```

Comment:
Why alias this? I was never a fan of how we alias stuff in `api` to the base `yarpc` package, what is the advantage?

Reply:
It's so we only need to have one import for tests.  Any test only needs to add this:
```
. "go.uber.org/yarpc/x/yarpctest"
```
and it gets all the functions it will ever need.

In this case, we need the "Action" and "Lifecycle" interfaces for how we define the testing struct:
```
tests := []struct {
	name     string
	services Lifecycle
	requests Action
} {
        {
	    name: "http to http request",
	    services: Lifecycles(...),
	    requests: Actions(...),
        },
},
```

Reply:
I don't feel strongly about this, so I defer to your judgement, my two cents is that an extra import isn't a big deal, and that defining two types is "messier" and can lead to more problems down the road. But I'm fine with either.

Reply:
for now I'll leave it as is, it's all in an /x/ package so we reserve the right to change it later if need be.